tests/chainer_tests/functions_tests/math_tests/test_sign.py

```diff
@@ -1,76 +1,56 @@
     'dtype': [numpy.float16, numpy.float32, numpy.float64],
 }))
-class TestSign(unittest.TestCase):
+@testing.fix_random()
+@testing.inject_backend_tests(
+    None,
+    # CPU tests
+    [
+        {},
+    ]
+    # GPU tests
+    + testing.product({
+        'use_cuda': [True],
+        'cuda_device': [0, 1],
+    })
+    # ChainerX tests
+    + [
+        {'use_chainerx': True, 'chainerx_device': 'native:0'},
+        {'use_chainerx': True, 'chainerx_device': 'cuda:0'},
+        {'use_chainerx': True, 'chainerx_device': 'cuda:1'},
+    ])
+class TestSign(testing.FunctionTestCase):
+
+    skip_backward_test = True
+    skip_double_backward_test = True
 
     def setUp(self):
-        self.x = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
-        self.gy = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
-        self.ggx = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
-
-        # Avoid non-differentiable point
-        self.x[(abs(self.x) < 1e-2)] = 1
-
-    def check_forward(self, x_data, xp):
-        x = chainer.Variable(x_data)
-        y = F.sign(x)
-        v = xp.sign(x_data)
-
-        assert x.data.dtype == y.data.dtype
-        testing.assert_allclose(v, y.data, atol=1e-7, rtol=1e-7)
-
-    def test_forward_cpu(self):
-        self.check_forward(self.x, numpy)
-
-    @attr.gpu
-    def test_forward_gpu(self):
-        self.check_forward(cuda.to_gpu(self.x), cuda.cupy)
+        self.check_forward_options.update({'atol': 1e-7, 'rtol': 1e-7})
 
-    def check_forward_ndarray(self, x_data, xp):
-        y = F.sign(x_data)
-        v = xp.sign(x_data)
+    def generate_inputs(self):
+        x = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
 
-        assert x_data.dtype == y.data.dtype
-        assert x_data.dtype == v.dtype
-        assert isinstance(y, chainer.Variable)
-        testing.assert_allclose(v, y.data, atol=1e-7, rtol=1e-7)
-
-    def test_forward_ndarray_cpu(self):
-        self.check_forward_ndarray(self.x, numpy)
-
-    @attr.gpu
-    def test_forward_ndarray_gpu(self):
-        self.check_forward_ndarray(cuda.to_gpu(self.x), cuda.cupy)
-
-    def check_backward(self, x_data, y_grad):
-        gradient_check.check_backward(
-            F.sign, x_data, y_grad)
-
-        # Explicitly check that gradients are `None`
-        x = chainer.Variable(x_data)
-        F.sign(x).backward()
-        assert x.grad is None
-
-    def test_backward_cpu(self):
-        self.check_backward(self.x, self.gy)
-
-    @attr.gpu
-    def test_backward_gpu(self):
-        self.check_backward(
-            cuda.to_gpu(self.x), cuda.to_gpu(self.gy))
+        # Avoid non-differentiable point
+        x[(abs(x) < 1e-2)] = 1
+        return x,
+
+    def forward(self, inputs, device):
+        x, = inputs
+        y = functions.sign(x)
+        return y,
+
+    def forward_expected(self, inputs):
+        x, = inputs
+        expected = numpy.sign(x)
+        expected = numpy.asarray(expected)
```

Comment:
Could you use `utils.force_array` instead of `numpy.asarray` here?
https://github.com/chainer/chainer/blob/d19022f59e25a95cdaf11d30ccb839740a2ca34d/chainer/utils/__init__.py#L38-L52

Reply:
Have made the required changes